xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -117,32 +118,56 @@ void CPVRClients::UpdateAddons(const std::string& changedAddonId /*= ""*/)
       AddonInfoPtr addon = addonWithStatus.first;
       bool bEnabled = addonWithStatus.second;
 
-      if (bEnabled && (!IsKnownClient(addon->ID()) || !IsCreatedClient(addon->ID())))
+      std::vector<uint32_t> instanceIds;
+      if (addon->SupportInstanceSettings())
```

Comment:
This reads wrong, although it works. What have settings to do with the addon creation? Should CAddon or wherever this method comes from get another new method named "SupportsMultipleInstances()". Implementation could be just to call "SupportsInstanceSettings ", but the new method explaining what we want to know here is a much more clean interface.